// meteor-react-autoform/lib/errors.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var listItems = function listItems(errors, style) {
  var mappedErrors = [];

  errors.map(function (error) {
    mappedErrors[error.name] = error.message;
  });

  return Object.keys(mappedErrors).map(function (key) {
    return _react2.default.createElement(
      'li',
      { key: key, style: style.li },
      mappedErrors[key]
    );
  });
}; /**
    * Created by josh.welham on 27/05/16.
    */

var Errors = function Errors(_ref) {
  var errors = _ref.errors;
  var style = _ref.style;
  var title = _ref.title;
  return _react2.default.createElement(
    'div',
    { style: style.container },
    _react2.default.createElement(
      'h3',
      { style: style.h3 },
      title
    ),
    _react2.default.createElement(
      'ul',
      { style: style.ul },
      listItems(errors, style)
    )
  );
};

Errors.propTypes = {
  errors: _react2.default.PropTypes.array.isRequired,
  style: _react2.default.PropTypes.object,
  title: _react2.default.PropTypes.string
};

Errors.defaultProps = {
  style: {
    container: {
      // background: 'green'
    },
    h3: {
      // background: 'red'
    },
    ul: {
      // background: 'purple'
    },
    li: {
      // background: 'yellow'
    }
  },
  title: 'There was an error submitting the form:'
};

exports.default = Errors;
